deepchecks/nlp/utils/text_embeddings.py

```diff
@@ -53,21 +105,74 @@ def calculate_builtin_embeddings(text: np.array, model: str = 'miniLM',
+            chunked_texts = []
+            chunk_lens = []
+            encoded_texts = []
+            max_doc_length = 0
+            for text_sample in list_of_texts:
+                tokens_in_sample = encode_text(text_sample, encoding_name=encoding_name)
+                tokens_per_sample = []
+                for chunk in iterate_batched(tokens_in_sample, chunk_length=max_tokens):
+                    chunked_texts.append(chunk)
+                    chunk_lens.append(len(chunk))
+                    tokens_per_sample += chunk
+                    max_doc_length = max(max_doc_length, len(tokens_per_sample))
+                    if long_doc_averaging == 'truncate':
+                        break
+                encoded_texts.append(tokens_per_sample)
+
+            if max_doc_length > max_tokens:
+                if long_doc_averaging == 'average+warn':
+                    warnings.warn(f'At least one document is longer than {max_tokens} tokens, which is the maximum '
+                                  f'context window handled by {model}. Maximal document length '
+                                  f'found is {max_doc_length} tokens. The document will be split into chunks and the '
+                                  f'embeddings will be averaged. To avoid this warning, set '
+                                  f'long_doc_averaging="average" or long_doc_averaging="truncate".')
+                elif long_doc_averaging == 'raise':
+                    raise ValueError(f'At least one document is longer than {max_tokens} tokens, which is the maximum '
+                                     f'context window handled by {model}. Maximal document '
+                                     f'length found is {max_doc_length} tokens. To avoid this error, set '
+                                     f'long_doc_averaging="average" or long_doc_averaging="truncate".')
+
+            batch_size = 500
+            chunk_embeddings_output = []
+            for sub_list in tqdm([chunked_texts[x:x + batch_size] for x in range(0, len(chunked_texts), batch_size)],
+                                 desc='Calculating Embeddings '):
+                chunk_embeddings_output.extend(_get_embedding_with_backoff(sub_list, model=model_name))
+            chunk_embeddings = [embedding['embedding'] for embedding in chunk_embeddings_output]
+
+            result_embeddings = []
+            idx = 0
+            for tokens_in_sample in encoded_texts:
+                text_embeddings = []
+                text_lens = []
+                while idx < len(chunk_lens) and sum(text_lens) < len(tokens_in_sample):
+                    text_embeddings.append(chunk_embeddings[idx])
+                    text_lens.append(chunk_lens[idx])
+                    idx += 1
+
+                if long_doc_averaging == 'nan' and len(text_embeddings) > 1:
+                    text_embedding = np.ones_like(text_embeddings[0]) * np.nan
```

Comment:
If I get this correctly, you still calculate the text embeddings for all the chunks even if you truncate, you just don't use it in the end? I don't think this is good, this wastes a lot of resources + the user pays for each API call.